sdks/java/io/jdbc/src/main/java/org/apache/beam/sdk/io/jdbc/JdbcIO.java

```diff
@@ -830,11 +838,60 @@ public void populateDisplayData(DisplayData.Builder builder) {
       inner.populateDisplayData(builder);
     }
 
+    private boolean hasStatementAndSetter() {
+      return inner.getStatement() != null && inner.getPreparedStatementSetter() != null;
+    }
+
     @Override
     public PDone expand(PCollection<T> input) {
+
+      if (input.hasSchema() && !hasStatementAndSetter()) {
+        Schema schema = input.getSchema();
+
+        inner = inner.withStatement(generateStatement(inner.getTable(), schema.getFields()));
+        inner = inner.withPreparedStatementSetter(((element, preparedStatement) -> {
+          // assuming we have PCollection of <Row>. We don't have Row then we will use toRowFunction().
+          // TODO: use toRowFunction() to convert other data types into Row
+          mapAccordingToSchema((Row) element, preparedStatement, schema); // Type casting element into Row is temporary untill we implement usage of toRowFunction()
+        }));
+      }
+
       inner.expand(input);
       return PDone.in(input.getPipeline());
     }
+
+    private void mapAccordingToSchema(Row element, PreparedStatement preparedStatement, Schema schema) {
+      List<Schema.Field> fields = schema.getFields();
+      IntStream.range(0, fields.size()).forEach((index) -> {
+        JdbcUtil.setFieldPreparedStatement(element, preparedStatement, fields.get(index).getType(), index);
+      });
+    }
```

Comment:
`mapAccordingToSchema()` method calls `preparedStatement.setInt(index, data)` and other methods on `preparedStatement`